flink-runtime/src/main/java/org/apache/flink/runtime/rpc/akka/AkkaInvocationHandler.java

```diff
@@ -189,7 +191,49 @@ public void stop() {
 		rpcEndpoint.tell(Processing.STOP, ActorRef.noSender());
 	}
 
-	// ------------------------------------------------------------------------
+	@Override
+	public boolean equals(Object o) {
+		if (this == o) {
+			return true;
+		}
+
+		if (o == null) {
+			return false;
+		}
+
+		if(Proxy.isProxyClass(o.getClass())) {
+			return o.equals(this);
+		}
```

Comment:
@tillrohrmann ,  It does not means that AkkaInvocationHandler be a proxy class. In fact it means if input parameter class is a proxy class, then return o.equals(this) result. Here is my reason.  when call connect method in AkkaRpcService, the returned gateway which is wrapped in Future is in fact a Proxy. When I call equals method to compare two gateway, the equals method of AkkaInvocationHandler will be called. But the input parameter is still another Gateway  which class is Proxy class instead of AkkaInvocationHandler.